tests/websocket_new_test.py

```diff
@@ -228,3 +229,245 @@ def error_detector(environ, start_response):
         sock.sendall(b'\x07\xff')  # Weird packet.
         done_with_request.wait()
         assert not error_detected[0]
+
+
+class TestWebSocketWithCompression(tests.wsgi_test._TestBase):
+    TEST_TIMEOUT = 5
+
+    def set_site(self):
+        self.site = wsapp
+
+    def setUp(self):
+        super(TestWebSocketWithCompression, self).setUp()
+        self.connect = '\r\n'.join([
+            "GET /echo HTTP/1.1",
+            "Upgrade: websocket",
+            "Connection: upgrade",
+            "Host: %s:%s" % self.server_addr,
+            "Origin: http://%s:%s" % self.server_addr,
+            "Sec-WebSocket-Version: 13",
+            "Sec-WebSocket-Key: d9MXuOzlVQ0h+qRllvSCIg==",
+            "Sec-WebSocket-Extensions: %s",
+            '\r\n'
+        ])
+        self.handshake_re = re.compile(six.b('\r\n'.join([
+            'HTTP/1.1 101 Switching Protocols',
+            'Upgrade: websocket',
+            'Connection: Upgrade',
+            'Sec-WebSocket-Accept: ywSyWXCPNsDxLrQdQrn5RFNRfBU=',
+            'Sec-WebSocket-Extensions: (.+)'
+            '\r\n',
+        ])))
+
+    @staticmethod
+    def get_deflated_reply(ws):
+        msg = ws._recv_frame(None)
+        msg.decompressor = None
+        return msg.getvalue()
+
+    def test_accept_basic_deflate_ext_13(self):
+        for extension in [
+            'permessage-deflate',
+            'PeRMessAGe-dEFlaTe',
+        ]:
+            sock = eventlet.connect(self.server_addr)
+
+            sock.sendall(six.b(self.connect % extension))
+            result = sock.recv(1024)
+
+            # The server responds the correct Websocket handshake
+            # print('Extension offer: %r' % extension)
+            match = re.match(self.handshake_re, result)
+            self.assertIsNotNone(match)
```

Comment:
`assert match is not None`
`assert len(match.groups()) == 1`

I know we still have other code using Java style asserts, please use assert statement in new code.